src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -902,6 +924,29 @@ public T GetItem(int index)
             }
         }
 
+        /// <summary>
+        /// Gets the item at the index if the index does not exceed the bounds of the list.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="index"/> is less than 0.</exception>
+        internal bool TryGetItem(int index, [MaybeNullWhen(false)] out T item)
+        {
+            this.SyncRoot.EnterReadLock();
+            try
+            {
+                if (index < this.content.Count)
```

Comment:
You should probably also add a index < 0 check here.

Reply:
`TryGetItem` is intended to throw `ArgumentOutOfRangeException` when `index` is negative, because that indicates developer error rather than the list changing size. That exception is documented above.